### ISSUE TITLE
Reproducibility Issues: Missing Scripts and Preprocessing Details for Brain-JEPA

### ISSUE THREAD

**Issue by jhpark-kaist** [Eric-LRL/Brain-JEPA#3]
Dear maintainers,

I’m trying to reproduce the results from **Brain-JEPA** using your official repository and released checkpoints, but I’m seeing substantially lower performance than reported. Both pre-training a new model and fine-tuning the provided checkpoint on our preprocessed data yield consistently worse results. We suspect the gap may stem from differences in data preprocessing and/or the exact loss function used. Could you please clarify the following?

---

### 1. UK Biobank 1K subset generation  
The paper refers to pretraining on a “UK Biobank 1K subset” created by a script named `make_ukbiobank1k`, but I can’t find this in the repo.  
- **Request:** Could you share the `make_ukbiobank1k` script or outline the exact steps to generate the UKB-1K dataset?

---

### 2. HCP-A downstream normalization  
The downstream scripts  
- `scripts/classification/run_downstream_LP_hca_sex.sh`  
- `scripts/classification/run_downstream_FT_hca_sex.sh`  

include a `--use_normalization` flag. In `src/datasets/hca_sex_datasets.py`, this implements:
```python
if self.use_normalization:
    mean = input_x.mean()
    std  = input_x.std()
    input_x = (input_x - mean) / std
```
—that is, sample-wise standard scaling across all ROIs and timesteps. However, **Section 3.1** of the paper states:

> “Robust scaling was implemented by subtracting the median and dividing by the interquartile range, calculated across participants for each ROI.”

- **Question:** Is the per-sample `(x – μ)/σ` normalization in the code intentional? 
- **If not:** Could you clarify the exact HCP-A preprocessing procedure used to obtain the published results?
- **If intentional:** Why is preprocessing handled differently between pre-training and downstream evaluation? Shouldn’t the preprocessing pipeline be unified to ensure consistency?

---

### 3. Loss function discrepancy  
In `src/train.py`, the downstream loss is defined as:
```python
loss = F.smooth_l1_loss(z, h)
```
—but **Eq. (6)** in the paper describes a mean-squared error over three frequency bands (α, β, γ):

$$
L = \frac{1}{3K} \sum_{r\in\{\alpha,\beta,\gamma\}} \|\hat{s}_r^y - s_r^y\|_2^2
$$

- **Which loss function was actually used in your official experiments?**  
- **If you used Smooth L1:** Could you explain how it relates to the formulation in Eq. (6)?

---

Thanks so much for sharing your great research on GitHub! It’s been a huge inspiration for my own work.
I’d really appreciate any scripts, details, or pointers you could share so I can reproduce your results faithfully—thanks in advance!

Best regards.

**Comment by Eric-LRL** [comment]
Hi @jhpark-kaist 

Thanks for taking the time to explore our paper in detail!

## Pre‑processing pipelines
We would like to highlight that all of the preprocessing details on UKB and HCA are documented in [Mansour _et al._](https://www.sciencedirect.com/science/article/pii/S105381192300558X) and [Wu _et al._](https://www.sciencedirect.com/science/article/pii/S105381192200684X?via%3Dihub), respectively, as noted in [README.md](https://github.com/Eric-LRL/Brain-JEPA/blob/main/README.md). For UKB, the preprocessed data from  [Mansour _et al._](https://www.sciencedirect.com/science/article/pii/S105381192300558X) can be directly downloaded from the official UKB platform and used in our model.

## 1. UK Biobank data loader
We just pushed the UKB loader to the repo. After pulling the latest commit, set `--root_dir` to the folder that holds the fMRI files of [Mansour _et al._](https://www.sciencedirect.com/science/article/pii/S105381192300558X) version you downloaded, specify the id‑split file in line 47, and everything should run out of the box.

## 2 & 3. Standardization vs. robust scaling
We actually evaluated both z‑score standardization and robust scaling. On HCP‑Sex, standardization may perform better. For the regression task, please consider using robust scaling.

The repo reflects the configs we ultimately used—treat that as the ground truth.

We guess your pretraining failure could come from preprocessing. Pinpointing the exact discrepancies in the preprocessing steps is challenging; therefore, we recommend that you can start by downloading the  [Mansour _et al._](https://www.sciencedirect.com/science/article/pii/S105381192300558X) version from the UKB platform to make the model work first. Once that succeeds, you can evaluate your own preprocessing further. Moreover, in our follow-up experiments, we indeed find that certain preprocessing variants could cause failures in this framework . And we are still exploring the underlying issues.

We recently performed some pretraining of Brain‑JEPA on new data splits for some analysis; the loss curve is attached below for your reference.

![Image](https://github.com/user-attachments/assets/635e6500-8711-4853-97df-451d7e6d5df9)

Hope this helps. Wish you a nice day!

Best,
Ruilin
